internal/versions/versions_test.go

```diff
@@ -650,9 +653,556 @@ var _ = Describe("GetReleaseImageByURL", func() {
 	})
 })
 
+var _ = Describe("SyncReleaseImages", func() {
+	var (
+		db          *gorm.DB
+		dbName      string
+		ctrl        *gomock.Controller
+		err         error
+		handler     *handler
```

Comment:
Why adding this dependency? Is it really so much better than stdlib [httptest](https://go.dev/src/net/http/httptest/example_test.go) ?

Reply:
I was now aware of httptest. I will try it out thanks

Reply:
I think `httpmock` is better for this case, as it doesn't actually make an `http` request, unlike `httptest`. If I want to use `httptest`, I will need to change the endpoint I make the request to configurable, but that is a bit odd as the endpoint is computed on the fly during execution (the query params). WDYT ?

Reply:
We are already looping through params to make up endpoint every time we want to make a request.
Not sure I see the issue here, in my mind it would be something like that:
```
var responseMatcher map[string]RequestResponseParameter

for _, parameters := range requestsResponsesParameters {
    endpoint := fmt.Sprintf(VersionsAPITemplate, parameters.Channel, parameters.Version, parameters.CPUArchitecture)
    responseMatcher[endpoint] = parameters
}

server = httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    if params, ok := responseMatcher[strings.TrimSpace(r.URL.Path)] {
         w.WriteHeader(http.StatusOK)
         w.Write(params.Response)
         return
    }
    w.WriteHeader(http.StatusNotFound)
}))
```

Maybe slightly more verbose (can be easily factored it out to something like `mockHTTPRequests(requestsResponsesParameters)`, but very simple and not requiring external dependency.
It would be also easy to mock errors as you'd just need to add an error element to the list (although I guess the same can be done with httpmock if we change approach slightly)
Does it make sense? WDYT?

Reply:
Got you now. Nice thank you